include/seqan3/alphabet/concept.hpp

```diff
@@ -149,11 +151,20 @@ template <typename t>
 concept alphabet_concept = semi_alphabet_concept<t> && requires (t t1, t t2)
 {
     // conversion to char
-    { to_char(t1) } -> underlying_char_t<std::remove_reference_t<t>>;
+    requires noexcept(to_char(t1));
+    requires std::Same<decltype(to_char(t1)), underlying_char_t<std::remove_reference_t<t>>>;
 
     // assignment from char
-    { assign_char(t1,  0) }                          -> std::remove_reference_t<t> &;
-    { assign_char(std::remove_reference_t<t>{}, 0) } -> std::remove_reference_t<t> &&;
+    requires noexcept(assign_char(t1,  0));
+    requires std::Same<decltype(assign_char(t1,  0)), std::remove_reference_t<t> &>;
+    requires noexcept(assign_char(std::remove_reference_t<t>{},  0));
+    requires std::Same<decltype(assign_char(std::remove_reference_t<t>{}, 0)), std::remove_reference_t<t> &&>;
+
+    // assignment from char (strict)
+//     { assign_char_strict(t1,  0) } -> noexcept;
```

Comment:
maybe add this when you add assign_char_strict but as out-commented code with no remark.